libraries/standard/http/include/http_client.h

```diff
@@ -0,0 +1,339 @@
+#ifndef HTTP_CLIENT_H_
+#define HTTP_CLIENT_H_
+
+#include <stdint.h>
+#include <stdbool.h>
+
+/**
+ * @brief Maximum amount of headers allowed from the server.
+ * 
+ * If the server sends more headers than this configuration, then status code
+ * #HTTP_SECURITY_ALERT_TOO_MANY_HEADERS is returned from #HTTPClient_Send.
+ */
+#ifndef HTTP_MAX_HEADERS_BYTES
+    #define HTTP_MAX_HEADERS_BYTES      2048U
+#endif
+
+/**
+ * @brief The HTTP header "User-Agent" value.
+ * 
+ * The following headerline is automatically written to
+ * #HTTPRequestHeaders_t.pBuffer:
+ * "User-Agent: AWS\r\n"
+ */ 
+#ifndef HTTP_USER_AGENT_VALUE
+    #define HTTP_USER_AGENT_VALUE       "AWS"
+#endif
+
+/**
+ * Supported HTTP request methods.
+ */
+#define HTTP_METHOD_GET                 "GET"  /**< HTTP Method GET. */
+#define HTTP_METHOD_PUT                 "PUT"  /**< HTTP Method GET. */
+#define HTTP_METHOD_POST                "POST" /**< HTTP Method GET. */
+#define HTTP_METHOD_HEAD                "HEAD" /**< HTTP Method GET. */
+
+
+/**
+ * Flags for #HTTPRequestInfo_t.flags.
+ * These headers control what headers are written or not to the 
+ * #HttpRequestHeaders_t.pBuffer.
+ */
+/**
+ * @brief Set this flag to indicate the request is for a persistent connection.
+ * 
+ * Setting this will cause a "Connection: Keep-Alive" to be written to the
+ * request.
+ * 
+ * Keep in mind some third-party servers do not have to honor persistent
+ * connection requests.
+ */
+#define HTTP_REQUEST_KEEP_ALIVE_FLAG                    0x1U
+/** 
+ * @brief Set this flag to disable automatically writing the Content-Length
+ * header. 
+ */
+#define HTTP_REQUEST_DISABLE_AUTO_CONTENT_LENGTH_FLAG   0x2U
+
+
+/**
+ * Flags for #HTTPResponse_t.flags.
+ * These flags are in #HTTPResponse_t.flags after #HTTPClient_Send() returns.
+ */
+ /**
+  * @brief This will be set to true if header "Connection: close" is found.
+  * 
+  * If a "Connection: close" header is present the client should always
+  * close the connection.
+  */
+#define HTTP_RESPONSE_CONNECTION_CLOSE_FLAG             0x1U
+/**
+ * @brief This will be set to true if header "Connection: Keep-Alive" is found.
+ */
+#define HTTP_RESPONSE_CONNECTION_KEEP_ALIVE_FLAG        0x2U
+
+/**
+ * @brief The HTTPNetworkContext is an incomplete type. The application must 
+ * define HTTPNetworkContext to the type of their network context. This context
+ * is passed into the network interface functions.
+ */
+struct HTTPNetworkContext;
+typedef struct HTTPNetworkContext * HTTPNetworkContext_t;
+
+/**
+ * @brief Transport interface for sending data over the network.
```

Comment:
Wasn't the decision to have a common network interface for MQTT & HTTP until there is a need to separate them out?

Reply:
The common network interface implementations are in the demos.

Reply:
It is simple enough to change this. Let's not dwell on it.